plat/common/include/uk/plat/common/memory.h

```diff
@@ -112,6 +112,87 @@ ukplat_memregion_list_insert(struct ukplat_memregion_list *list,
 	return (int)i;
 }
 
+/**
+ * Insert a new region into the memory region list. This extends
+ * ukplat_memregion_list_insert to carve out the area of any pre-existing
+ * overlapping regions. The virtual addresses will leave out any carved out
+ * region.
+ * If there is not enough space for all resulting regions, the function will
+ * insert as many as possible and then signal return an error.
+ * @param list
+ *   The memory region list to insert the range into
+ * @param mrd
+ *   The memory range to insert
+ * @param min_size
+ *   The minimum size a inserted region has to have. Regions smaller than this
+ *   size won't be added. Settings this parameter zero will disable this
+ *   behavior.
+ * @returns
+ *   Zero if the operation was successful, a negative errno otherwise
+ */
+static inline int
+ukplat_memregion_list_insert_split_phys(struct ukplat_memregion_list *list,
+					const struct ukplat_memregion_desc *mrd,
+					const __sz min_size)
+{
+	struct ukplat_memregion_desc *mrdp;
+	struct ukplat_memregion_desc mrdc;
+	__paddr_t pstart, pend;
+	__vaddr_t voffset;
+	int i;
+	int rc;
+
+	voffset = mrd->vbase - mrd->pbase;
+	pstart = mrd->pbase;
+	pend = mrd->pbase + mrd->len;
+
+	mrdc = *mrd;
+
+	/* TODO: The following code does not make use of the tracked iteration
+	 * index to insert elements at the correct location and instead uses the
+	 * generic insertion routine. For large memory region lists this could
+	 * be potentially slow.
+	 */
+	for (i = 0; i < (int)list->count; i++) {
+		mrdp = &list->mrds[i];
+		if (!ukplat_memregion_desc_overlap(mrdp, pstart, pend))
+			continue;
+		if (pend <= mrdp->pbase)
+			break;
+
+		if (!mrdp->type)
+			continue;
+
+		if (pstart < mrdp->pbase) {
+			/* Some part of the inserted region is before the
+			 * overlapping region. Try to insert that part if it's
+			 * large enough.
+			 */
+			mrdc.pbase = pstart;
+			mrdc.vbase = pstart + voffset;
+			mrdc.len   = mrdp->pbase - pstart;
+
+			if (mrdc.len > min_size) {
```

Comment:
According to the function description it should be:
```suggestion
			if (mrdc.len >= min_size) {
```